src/components/safe-apps/SafeAppsSignMessageModal/ReviewSafeAppsSignMessage.tsx

```diff
@@ -0,0 +1,132 @@
+import { ReactElement, useMemo } from 'react'
+import { hashMessage, _TypedDataEncoder } from 'ethers/lib/utils'
+import { Box } from '@mui/system'
+import { TextField, Typography } from '@mui/material'
+import WarningIcon from '@mui/icons-material/Warning'
+import { isObjectEIP712TypedData, Methods } from '@gnosis.pm/safe-apps-sdk'
+import { OperationType, SafeTransaction } from '@gnosis.pm/safe-core-sdk-types'
+
+import SendFromBlock from '@/components/tx/SendFromBlock'
+import { InfoDetails } from '@/components/transactions/InfoDetails'
+import EthHashInfo from '@/components/common/EthHashInfo'
+import SignOrExecuteForm from '@/components/tx/SignOrExecuteForm'
+import { generateDataRowValue } from '@/components/transactions/TxDetails/Summary/TxDataRow'
+import { SafeAppsSignMessageParams } from '../SafeAppsSignMessageModal'
+import useChainId from '@/hooks/useChainId'
+import useAsync from '@/hooks/useAsync'
+import useSafeInfo from '@/hooks/useSafeInfo'
+import { getSignMessageLibDeploymentContractInstance } from '@/services/contracts/safeContracts'
+import { createTx } from '@/services/tx/txSender'
+import { convertToHumanReadableMessage } from '../utils'
+import { dispatchSafeAppsTx } from '@/services/tx/txSender'
+
+type ReviewSafeAppsSignMessageProps = {
+  safeAppsSignMessage: SafeAppsSignMessageParams
+}
+
+const ReviewSafeAppsSignMessage = ({
+  safeAppsSignMessage: { message, method, requestId },
+}: ReviewSafeAppsSignMessageProps): ReactElement => {
+  const { safe } = useSafeInfo()
+  const chainId = useChainId()
+
+  const isTextMessage = method === Methods.signMessage && typeof message === 'string'
+  const isTypedMessage = method === Methods.signTypedMessage && isObjectEIP712TypedData(message)
+
+  const signMessageDeploymentInstance = useMemo(() => getSignMessageLibDeploymentContractInstance(chainId), [chainId])
+  const signMessageAddress = signMessageDeploymentInstance.address
+
+  const canExecute = safe.threshold === 1
+
+  const readableData = useMemo(() => {
+    if (isTextMessage) {
+      return convertToHumanReadableMessage(message)
+    } else if (isTypedMessage) {
+      return JSON.stringify(message, undefined, 2)
+    }
+  }, [isTextMessage, isTypedMessage, message])
+
+  const [safeTx, safeTxError] = useAsync<SafeTransaction>(() => {
+    let txData
+
+    if (isTextMessage) {
+      txData = signMessageDeploymentInstance.interface.encodeFunctionData('signMessage', [hashMessage(message)])
+    } else if (isTypedMessage) {
+      txData = signMessageDeploymentInstance.interface.encodeFunctionData('signMessage', [
+        _TypedDataEncoder.hash(message.domain, message.types, message.message),
+      ])
+    }
+
+    return createTx({
+      to: signMessageAddress,
+      value: '0',
+      data: txData || '0x',
+      operation: OperationType.DelegateCall,
+    })
+  }, [])
```

Comment:
We might be missing some dependencies here as well e.g. `message`

Reply:
You are right!